src/components/camera/server.cpp

```diff
@@ -0,0 +1,162 @@
+#include <components/camera/server.hpp>
+#include <components/camera/camera.hpp>
+#include <rpc/server.hpp>
+#include <gen/google/api/http.pb.h>
+
+::grpc::Status CameraServer::DoCommand(
+    ::grpc::ServerContext* context,
+    const ::viam::common::v1::DoCommandRequest* request,
+    ::viam::common::v1::DoCommandResponse* response) {
+
+    std::shared_ptr<ResourceBase> rb = sub_svc->resource(request->name());
+    if (rb == nullptr) {
+        return grpc::Status(grpc::UNKNOWN, "resource not found: " + request->name());
+    }
+
+    std::shared_ptr<Camera> camera = std::dynamic_pointer_cast<Camera>(rb);
+    std::shared_ptr<std::unordered_map<std::string, std::shared_ptr<ProtoType>>> command = struct_to_map(request->command());
```

Comment:
(nit) I think we can get away with not defining this `command` here at all and in the below line just call `camera->do_command(request->name(), request->command());`